_guides/managing-oauth-clients.md

```diff
@@ -0,0 +1,209 @@
+---
+title: Managing OAuth Clients
+layout: article
+section: Developing Components
+order: 1
+since: 20190403
+---
+This document reveals the new approach to OAuth utilization within the platform
+and explains how to [manage](#managing-oauth-client) OAuth clients for
+components in a tenant.
+
+## Changes
+
+
+OAuth client was defined for a component in its environment variables. This way
+global access to the component could not be enabled, because users in other tenants would see the
+component and its keys. Now OAuth clients are defined
+separately from the components’ environment variables, allowing the users in the
+other tenant to see all available components without security breach.
+
+To manage OAuth clients the user requires the following permissions:
+
+
+`tenants.oauth_clients.get`
+
+`tenants.oauth_clients.edit`
+
+`tenants.oauth_clients.create`
+
+`tenants.oauth_clients.delete`
+
+
+To acquire these permissions, please contact support.
+
+## Managing OAuth Client
+
+To initiate work with OAuth client, first specify `"useOAuthClient": true` in the _component.json_ file. For compatibility reasons, we should also create a link between the new implementation and environment variables if they were present.
+
+**EXAMPLE:**
+
+If you had the following defined for environment variables:
+```
+"envVars": {
+    "PIZZAVAN_KEY": {
+      "required": true,
+      "description": "Your Pizza Van OAuth client key"
+    },
+    "PIZZAVAN_SECRET": {
+      "required": true,
+      "description": "Your Pizza Van OAuth client secret"
+    }
+  }
+```
+Add the following object to _component.json_:
+```
+"oauth2": {
+      "client_id": "{{PIZZAVAN_KEY}}",
+      "client_secret": "{{PIZZAVAN_SECRET}}",
+      "auth_uri": "https://auth_uri",
+      "token_uri": "https://token_uri"
+    }
+```
+
+OAuth client management includes the following actions: create,
+[retrieve](#to-retrieve-oauth-clients-in-a-tenant-we),
+[update](#to-update-oauth-clients-in-a-tenant-we-will-use-the-following-api-request)
+and [delete](#to-delete-an-oauth-client-we-will-use-the-following-api-request).
+
+#### To create OAuth clients, we will use the following API request:
```

Comment:
I think it shouldn't be a paragraph title but a sentence guiding the user. Something like this.

> Now that you specified that a component requires an Auth Client to function, let's explore how to create these clients using the REST API.

Reply:
We also need to mention what that a tenant `X` won't be able to use the component `C` made `global` by tenant `Y` until `X` provided a client for `C`. @anton-kotenko will provide you more details on that.

Reply:
Additional details: 
1. Component needs oauth client to be defined to work correctly
2. Oauth clients lives in tenants (e.g. oauth client may be defined in different tenants but for same component. And component will be used oauth-client from the tenant, that contains flow, that  contains component as one of steps
3. So to avoid situation when component won't work, we need to hide component in those tenants, where there are no proper oauth client.

Let's look at scenario:
1. we (E.io) created new super-puper component
2. we debugged in our tenant and want to publish it global (to make it visible and usable in all tenants)
3. we publish it
4. But component is not really visible in all other tenants, until tenant define proper oauth client for component

Reply:
With the lv4 title I'm trying to highlight the different actions, because otherwise their contents are really difficult to distinguish. Also, as I understand, markdown only uses headings as anchors for internal linking. As for the wording, it may be adopted, but I'll have to revise all the other docs according to the style.